python/lsst/ts/ess/common/device/mock_device.py

```diff
@@ -0,0 +1,188 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MockDevice"]
+
+import asyncio
+import logging
+import random
+from typing import Callable
+
+from .base_device import BaseDevice
+from ..constants import (
+    DISCONNECTED_VALUE,
+    MockDewPointConfig,
+    MockHumidityConfig,
+    MockPressureConfig,
+    MockTemperatureConfig,
+)
+from ..sensor import BaseSensor, Hx85aSensor, Hx85baSensor, TemperatureSensor
+
+
+class MockDevice(BaseDevice):
+    """Mock Sensor Device.
+
+    Parameters
+    ----------
+    name: `str`
+        The name of the device.
+    device_id: `str`
+        The hardware device ID to connect to.
+    sensor: `BaseSensor`
+        The sensor that produces the telemetry.
+    callback_func : `Callable`
+        Callback function to receive the telemetry.
+    log: `logging.Logger`
+        The logger to create a child logger for.
+    disconnected_channel: `int`, optional
+        The channel number for which this class will mock a disconnection.
+    missed_channels: `int`, optional
+        The number of channels to not output to mock connecting to the sensor
+        in the middle of receiving data from it.
+    in_error_state: `bool`, optional
+        The sensor produces an error (True) or not (False) when being read.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        device_id: str,
+        sensor: BaseSensor,
+        callback_func: Callable,
+        log: logging.Logger,
+        disconnected_channel: int = None,
+        missed_channels: int = 0,
+        in_error_state: bool = False,
+    ) -> None:
+        super().__init__(
+            name=name,
+            device_id=device_id,
+            sensor=sensor,
+            callback_func=callback_func,
+            log=log,
+        )
+        self._disconnected_channel = disconnected_channel
+        self._missed_channels = missed_channels
+        self._in_error_state = in_error_state
+
+    async def basic_open(self) -> None:
+        """Open the Sensor Device."""
+        pass
+
+    def _format_temperature(self, i: int) -> str:
+        """Creates a formatted string representing a temperature for the given
+        channel.
+
+        Parameters
+        ----------
+        i: `int`
+            The 0-based temperature channel.
+        Returns
+        -------
+        s: `str`
+            A string representing a temperature.
+        """
+        if i < self._missed_channels:
+            return ""
+
+        prefix = f"C{i:02d}="
+        value = random.uniform(MockTemperatureConfig.min, MockTemperatureConfig.max)
+        if i == self._disconnected_channel:
+            value = float(DISCONNECTED_VALUE)
+        return f"{prefix}{value:09.4f}"
```

Comment:
I agree with your concern. The code was basically moved from ts_ess_controller to this project and was left untouched otherwise. Would you be OK with it if I opened another ticket to implement this and your other concern further down?

Reply:
I decided to do this in this ticket anyway.

Reply:
I decided to do this in this ticket anyway.